zne/noise_amplification/folding_amplifier/glorious_folding_amplifier.py

```diff
@@ -13,14 +13,16 @@
 """Glorious DAG Folding Noise Amplification (Temporary)"""
 import copy
 
-from qiskit.circuit.library import Barrier
 from qiskit.converters import circuit_to_dag, dag_to_circuit
 from qiskit.dagcircuit import DAGCircuit
 
 
 def _inverse_dag(dag_to_inverse: DAGCircuit) -> DAGCircuit:
     """Inverts dag circuit"""
-    dag_to_inverse = dag_to_circuit(dag_to_inverse).inverse()
+    dag_to_inverse = dag_to_circuit(dag_to_inverse)
+    dag_to_inverse.barrier()
+    dag_to_inverse = dag_to_inverse.inverse()
+    dag_to_inverse.barrier()
```

Comment:
Uhh, I had not though of this! Interesting!

You should refrain from this approach for several reasons, but probably the most important is that you are not doing what the function promises. Instead you are doing extra things (i.e. adding two barriers) that are not expected, this is sometimes referred to with the broader term _side effect_ which good code tries to avoid at all cost, as it makes it harder to understand and predict.

How about something like:
```python
barrier = Barrier(noisy_dag.num_qubits())
...
for _ in range(num_foldings):
    noisy_dag.apply_operation_back(barrier, qargs=noisy_dag.qubits)
    noisy_dag.compose(inverse_dag, inplace=True)
    ...
```